alembic/env.py

```diff
@@ -22,7 +22,7 @@
 
 # Set DB url from config
 settings = get_app_settings()
-config.set_main_option("sqlalchemy.url", settings.DB_URL)
+config.set_main_option("sqlalchemy.url", settings.DB_URL.replace("%", "%%"))
```

Comment:
Note that the PostgresProvider.db_url will always safely quote the password. So that means there will only ever be a single % in a row. as % turns into %25.

Users just provide the actual password in their configs. no escaping or special sequences needed. Rather, if you use those things, your config will break.

There is a reason many guides steer people away from using special characters in db passwords and instead opt for 32-64 character alphanumeric strings.